api/v1alpha/src/lib/metaManager.js

```diff
@@ -140,11 +155,16 @@ async function performDatasetMetadataUpdate(projectId, datasetId, accounts) {
  * @param  {} accounts
  */
 async function performTableMetadataUpdate(projectId, datasetId, tableId, accounts) {
-    const accessTypes = ["user", "group"];
-    const viewerRole = 'roles/bigquery.dataViewer';
     console.log(`Begin metadata update for table: ${datasetId}.${tableId}`);
-    let isDirty = false;
     const bigqueryUtil = new BigQueryUtil(projectId);
+    const exists = await bigqueryUtil.tableExists(datasetId, tableId);
+    if (!exists) {
+        console.warn(`Skipping metadata update for non-existant table: ${datasetId}.${tableId}`);
+        return false;
+    }
+    const accessTypes = ["user", "group", "serviceAccount"];
```

Comment:
What is the difference between **serviceAccount** and **user** accessTypes if both are **userByEmail** in BQ metadata? I could not interpret why this additional accessType is required... is there a bug/feature request I can check?

Reply:
Within BQ Dataset metadata there is no difference, a service account and a user both fall under 'userByEmail'. Group emails fall under 'groupByEmail'. BQ tables are different as they use IAM bindings, so the values there are group, user, and serviceAccount.